src/main/java/net/dries007/tfc/util/ICollapsableBlock.java

```diff
@@ -5,8 +5,88 @@
 
 package net.dries007.tfc.util;
 
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.dries007.tfc.objects.blocks.stone.BlockRockVariantFallable;
+import net.dries007.tfc.objects.blocks.wood.BlockSupport;
+
 // For raw stone, because it collapses
 public interface ICollapsableBlock
 {
-    // todo: this
+    // Used to get cobblestone variant of this collapsable raw stone
+    BlockRockVariantFallable getFallingVariant();
+
+    // Check if this pos is collapsable
+    default boolean canCollapse(World world, BlockPos pos)
+    {
+        return world.getBlockState(pos.down()).getMaterial().isReplaceable() && !BlockSupport.isBeingSupported(world, pos);
+    }
+
+    /**
+     * Check an area of blocks for collapsing mechanics
+     *
+     * @param worldIn the worldObj this block is in
+     * @param pos     the BlockPos this block has been mined from
+     */
+    default void checkCollapsingArea(World worldIn, BlockPos pos)
+    {
+        if (!worldIn.isAreaLoaded(pos.add(-32, -32, -32), pos.add(32, 32, 32)))
+            return; //First, let's check if this area is loaded
+        if (worldIn.rand.nextInt(100) < 10) //Then, we check rng if a collapse should trigger
+        {
+            //Rng the radius
+            int radX = (worldIn.rand.nextInt(5) + 4) / 2;
+            int radY = (worldIn.rand.nextInt(3) + 2) / 2;
+            int radZ = (worldIn.rand.nextInt(5) + 4) / 2;
+            for (BlockPos checking : BlockPos.getAllInBox(pos.add(-radX, -radY, -radZ), pos.add(radX, radY, radZ))) //9x5x9 max
+            {
+                //Check the area for a block collapse!
+                if (worldIn.getBlockState(checking).getBlock() instanceof ICollapsableBlock)
```

Comment:
Add a `Constants.RNG.nextFloat() < ConfigTFC.GENERAL.collapseChance` in this check